gammapy/time/plot.py

```diff
@@ -30,3 +32,131 @@ def plot_time_difference_distribution(time, ax=None):
+
+    fermi_cat = fetch_fermi_catalog('3FGL')
+
+    catalog_index = np.where(fermi_cat[1].data['Source_Name'] == name_3fgl)[0][0]
+
+    time_index_start = np.where(fermi_cat[3].data['Hist_Start'] >= fermi_met_start)[0][0]
+
+    # The final entry is the end of the last bin, so no off by one error
+    time_index_end = np.where(fermi_cat[3].data['Hist_Start'] <= fermi_met_end)[0][-1] + 1
+
+    time_start = fermi_cat[3].data['Hist_Start'][time_index_start: time_index_end]
+    time_end = np.roll(time_start, -1)
+
+    time_diff = 0.5 * (time_end - time_start)
+
+    # Trim because there is one more bin edge than there is bin mid point
+    time_diff = time_diff[0:-1]
+
+    # Midpoints of each bin.
+    time_mid = time_start[0:-1] + time_diff
+
+    cat_row = fermi_cat[1].data[catalog_index]
+
+    flux_history = cat_row['Flux_History'][time_index_start: time_index_end]
+
+    flux_history_lower_bound = cat_row['Unc_Flux_History'][time_index_start: time_index_end, 0]
+    flux_history_upper_bound = cat_row['Unc_Flux_History'][time_index_start: time_index_end, 1]
+    flux_history_lower_bound = abs(flux_history_lower_bound)
+
+    time_mid = (TIME_REF_FERMI + astropy.time.TimeDelta(time_mid, format='sec'))
+
+    time_at_bin_start = time_mid - astropy.time.TimeDelta(time_diff, format='sec')
+
+    time_at_bin_end = time_mid + astropy.time.TimeDelta(time_diff, format='sec')
+
+    time_mid = time_mid.plot_date
+
+    time_at_bin_start = time_at_bin_start.plot_date
+
+    time_at_bin_end = time_at_bin_end.plot_date
+
+    time_diff_at_bin_start = time_mid - time_at_bin_start
+
+    time_diff_at_bin_end = time_at_bin_end - time_mid
+
+    # Where a lower bound was recorded.
+    idx1 = np.where(np.isnan(flux_history_lower_bound) == False)
+
+    # Where a lower bound was not recorded.
+    idx2 = np.where(np.isnan(flux_history_lower_bound))
+
+    # Where no lower bound was recorded, set to zero flux.
+    flux_history_lower_bound[idx2] = flux_history[idx2]
+
+    # Plot data points and upper limits.
+    ax.errorbar(time_mid[idx1], flux_history[idx1],
+                 yerr=(flux_history_lower_bound[idx1], flux_history_upper_bound[idx1]),
+                 xerr=(time_diff_at_bin_start[idx1], time_diff_at_bin_end[idx1]),
+                 marker='o', elinewidth=1, linewidth=0, color='black')
+    ax.errorbar(time_mid[idx2], flux_history[idx2],
+                 yerr=(flux_history_lower_bound[idx2], flux_history_upper_bound[idx2]),
+                 marker=None, elinewidth=1, linewidth=0, color='black')
+    ax.scatter(time_mid[idx2], (flux_history[idx2] + flux_history_upper_bound[idx2]),
+                marker='v', color='black')
+    ax.set_xlabel('date')
+    ax.set_ylabel('flux [ph/cm^2/s]')
+    ax.set_ylim(ymin=0)
+    ax.xaxis_date()
+    ax.xaxis.set_major_formatter(mdates.DateFormatter('%m/%Y'))
+    ax.xaxis.set_major_locator(mdates.MonthLocator(interval=6))
+    plt.gcf().autofmt_xdate()
```

Comment:
To stay fully object-oriented and avoid side effects, you should access the correct figure via the axes:
```python
ax.figure.autofmt_xdate()
```